pkg/syncthing/syncthing.go

```diff
@@ -362,6 +369,28 @@ func (s *Syncthing) WaitForPing(ctx context.Context, local bool) error {
 	}
 }
 
+// IdentifyReadinessIssue attempts to identify the issue that is preventing syncthing from being ready
+func (s *Syncthing) IdentifyReadinessIssue(fs afero.Fs) error {
+	if s.RegexMatchesLogs(fs, regexp.MustCompile("Error opening database: mkdir .*: no space left on device")) {
```

Comment:
2 things to be aware:
* `MustCompile` will throw a panic if the expression cannot be compiled. As it is a static string I don't see a problem
* As the expression we pass to `MustCompile` is static you could create a global variable with the regex, so it is only compiled once.

I think given the use case of the CLI and this code both things are fine, but commenting them as you have more context about it and it case it might be relevant

Reply:
Thanks Nacho, I am aware of `MustCompile` and added test on-purpose to make sure we notice if it panics.  I've moved them at the top of the file (personally, I prefer the neglectable operation cost of compiling over having 2 vars defined ~200 lines of code above the line where they are used, considering they are not shared with any other function either, it makes the code less readable to me)